lesson4/argirovatina_hometask.rb

```diff
@@ -0,0 +1,38 @@
+class LazyHash < Hash
+
+  def method_missing(method_name)
+    key = get_correct_key(method_name)
+    self[key] = self[key].call if lazy?(key)
```

Comment:
Do you understand that it returns `nil` if `key` points on normal value, not `proc`?
See,
```ruby
hash[:key_with_regular_value] = 6
p hash.key_with_regular_value # => nil
```
Why does it happen? Because you last condition is treated by ruby interpreter as
```ruby
if lazy?(key)
  self[key] = self[key].call
else
  nil
end
```